feat: add TypeScript support and workspace configuration

adispring opened the pull request:
- Add TypeScript type definitions for chrome extension

- Configure pnpm workspace for better dependency management

- Add code quality tools (prettier, commitlint, eslint)

- Reorganize package dependencies

adispring committed:
feat: add TypeScript support and workspace configuration

- Add TypeScript type definitions for chrome extension

- Configure pnpm workspace for better dependency management

- Add code quality tools (prettier, commitlint, eslint)

- Reorganize package dependencies

tedjames commented:
Amazing!! Great work @adispring  👍 

I'm wrapping up some other work right now - let me review and test it on my machine tomorrow! We will get this + a few other updates in for a v1.3 release this weekend hopefully  :)

Appreciate your support very much. Thank you 🙏

adispring committed:
chore: clean up code style and add ESLint configuration for Chrome extension

- Remove .DS_Store file
- Add ESLint configuration specific to Chrome extension
- Refactor background.js with consistent code style (single quotes, formatting)

adispring committed:
refactor: remove unused TypeScript type definitions for background and devtools
